From 16dbcea4cb336cce711d83b33647ed31285f4677 Mon Sep 17 00:00:00 2001
From: sicherist <sichersicher@t-online.de>
Date: Sun, 7 Jan 2018 13:59:26 +0100
Subject: [PATCH 077/249] wiki: sony: Document pollux and pollux_windy unlock
 check

 * Document the access to the service menu for the Xperia
    Z Tablet (pollux) and Z Tablet WiFi (pollux_windy)
    as these devices have no dialer installed by default

 * Instructions based upon Sony's unlock methods:
    https://developer.sonymobile.com/unlockbootloader/

 * Introduced an optional device flag "install_variant"
    containing the "sony_unlock_contacts" key

Change-Id: I940113b31d685987ddbb11aa005e3b990f7c54ee
Signed-off-by: Adrian DC <radian.dc@gmail.com>
---
 _data/devices/pollux.yml                              | 1 +
 _data/devices/pollux_windy.yml                        | 1 +
 _includes/templates/recovery_install_fastboot_sony.md | 8 ++++++++
 3 files changed, 10 insertions(+)

diff --git a/_data/devices/pollux.yml b/_data/devices/pollux.yml
index ffce1a5..ce2f3cb 100644
--- a/_data/devices/pollux.yml
+++ b/_data/devices/pollux.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 172 mm (6.77 in)
 image: pollux.png
 install_method: fastboot_sony
+install_variant: sony_unlock_contacts
 kernel: android_kernel_sony_apq8064
 maintainers: [CaHbKaUp, CyberWalkMaN, Daedroza, Chippa_a, Adrian DC]
 name: Xperia Tablet Z LTE
diff --git a/_data/devices/pollux_windy.yml b/_data/devices/pollux_windy.yml
index 6dccab4..5e4705c 100644
--- a/_data/devices/pollux_windy.yml
+++ b/_data/devices/pollux_windy.yml
@@ -18,6 +18,7 @@ has_recovery_partition: true
 height: 172 mm (6.77 in)
 image: pollux.png
 install_method: fastboot_sony
+install_variant: sony_unlock_contacts
 kernel: android_kernel_sony_apq8064
 maintainers: [CaHbKaUp, CyberWalkMaN, Daedroza, Chippa_a, Adrian DC]
 name: Xperia Tablet Z Wi-Fi
diff --git a/_includes/templates/recovery_install_fastboot_sony.md b/_includes/templates/recovery_install_fastboot_sony.md
index 347479c..05078dd 100644
--- a/_includes/templates/recovery_install_fastboot_sony.md
+++ b/_includes/templates/recovery_install_fastboot_sony.md
@@ -6,6 +6,14 @@ This also includes your DRM keys, which are stored in the Trim Area partition (a
 Before proceeding, ensure the data you would like to retain is backed up to your PC and/or your Google account, or equivalent.
 If you wish to backup the TA partition first, you can find tutorials related to your device on the internet." %}
 
+{% if device.install_variant and device.install_variant contains "sony_unlock_contacts" %}
+1. Open the `Contacts` app.
+2. Click on the menu button in the top right.
+3. Click on `Settings`.
+4. Long press (for about 5 seconds) somewhere in the empty space just above navigation bar, then release your finger. A button named `Enter service code` will then appear in the same area (this may take a few attempts).
+5. Click on `Enter service code`. A dialer will appear.
+{% endif %}
+
 1. On the device, dial `*#*#7378423#*#*` (`*#*#SERVICE#*#*`) to launch the service menu.
 2. Go to `service info` > `configuration` and check `rooting status` - you can only continue if it says `Bootloader unlock allowed: Yes`.
 3. Connect the device to your PC via USB.
-- 
2.17.1

